lib/bundler/cli/info.rb

```diff
@@ -50,10 +50,17 @@ def print_gem_path(spec)
     end
 
     def print_gem_info(spec)
+      metadata = spec.respond_to?(:metadata) ? spec.metadata : {}
```

Comment:
Since our `required_rubygems_version` is `>= 2.5.2`, maybe we don't need this `respond_to?`?

Reply:
That makes sense to me. I was a little defensive when I wrote this, as I'm not too familiar with the codebase.

The `spec` here is an instance of `Bundler::StubSpecification`. So the `metadata` method comes in via method missing:

https://github.com/bundler/bundler/blob/fcb0fef62096bd8d64383f5bbfa584ce77f40fd8/lib/bundler/remote_specification.rb#L99-L107

`_remote_specification` is a `Gem::Specification` in all the experiments I've performed locally. I'm unaware of any scenario where it would be an instance of a different class.

I'll remove the check.